macros/src/lib.rs

```diff
@@ -102,52 +102,22 @@ fn gen_bundle_impl(
 ) -> TokenStream2 {
     let num_tys = tys.len();
     let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
-    let with_static_ids_inner = quote! {
-        {
-            let mut dedup = ::std::collections::HashSet::new();
-            for &(ty, name) in [#((::std::any::TypeId::of::<#tys>(), ::std::any::type_name::<#tys>())),*].iter() {
-                if !dedup.insert(ty) {
-                    ::std::panic!("{} has multiple {} fields; each type must occur at most once!", stringify!(#ident), name);
-                }
-            }
-
-            let mut tys = [#((::std::mem::align_of::<#tys>(), ::std::any::TypeId::of::<#tys>())),*];
-            tys.sort_unstable_by(|x, y| {
-                ::std::cmp::Ord::cmp(&x.0, &y.0)
-                    .reverse()
-                    .then(::std::cmp::Ord::cmp(&x.1, &y.1))
-            });
-            let mut ids = [::std::any::TypeId::of::<()>(); #num_tys];
-            for (id, info) in ::std::iter::Iterator::zip(ids.iter_mut(), tys.iter()) {
-                *id = info.1;
-            }
-            ids
-        }
-    };
-    let with_static_ids = if generics.params.is_empty() {
-        quote! {
+    quote! {
+        impl #impl_generics ::hecs::Bundle for #ident #ty_generics #where_clause {
             #[allow(non_camel_case_types)]
             fn with_static_ids<__hecs__T>(f: impl ::std::ops::FnOnce(&[::std::any::TypeId]) -> __hecs__T) -> __hecs__T {
-                ::hecs::lazy_static::lazy_static! {
-                    static ref ELEMENTS: [::std::any::TypeId; #num_tys] = {
-                        #with_static_ids_inner
-                    };
+                let mut tys = [#((::std::mem::align_of::<#tys>(), ::std::any::TypeId::of::<#tys>())),*];
+                tys.sort_unstable_by(|x, y| {
+                    ::std::cmp::Ord::cmp(&x.0, &y.0)
+                        .reverse()
+                        .then(::std::cmp::Ord::cmp(&x.1, &y.1))
+                });
+                let mut ids = [::std::any::TypeId::of::<()>(); #num_tys];
+                for (id, info) in ::std::iter::Iterator::zip(ids.iter_mut(), tys.iter()) {
+                    *id = info.1;
                 }
-
-                f(&*ELEMENTS)
+                f(&ids)
             }
-        }
-    } else {
-        quote! {
-            #[allow(non_camel_case_types)]
-            fn with_static_ids<__hecs__T>(f: impl ::std::ops::FnOnce(&[::std::any::TypeId]) -> __hecs__T) -> __hecs__T {
-                f(&#with_static_ids_inner)
-            }
-        }
-    };
-    quote! {
-        impl #impl_generics ::hecs::Bundle for #ident #ty_generics #where_clause {
-            #with_static_ids
```

Comment:
I'd (weakly) prefer to retain the separate case for monomorphic structs, so this change doesn't slow down existing code.

Reply:
I don't quite get what you mean. Are you saying you want to retain the lazy_static check in the non generic version? Because otherwise now the snippets for both generic and non-generic structs are the same.